src/helpers/reservations.ts

```diff
@@ -281,167 +283,130 @@ const computeCardUsage = (person: MockPerson) => {
-      cardDetails.cardPresentLimits.daily.max_amount_cents
-    ) {
-      await triggerCardDeclinedWebhook(
-        cardAuthorizationDeclined,
-        CardAuthorizationDeclineReason.CARD_PRESENT_AMOUNT_LIMIT_REACHED_DAILY,
-        person
-      );
-      throw new Error(
-        `Daily card_present amount limit exceeded (${dailyLimitAfterAuthorization} > ${cardDetails.cardPresentLimits.daily.max_amount_cents})`
-      );
-    }
+  const { amount: monthlyATMLimitAmount, count: monthlyATMLimitCount } =
+    getCardLimits(
+      cardData.controls,
+      CardSpendingLimitPeriod.MONTHLY,
+      DimensionType.ATM_WITHDRAWAL
+    );
 
-    if (
-      currentCardUsage.cardPresent.daily.transactions >
-      cardDetails.cardPresentLimits.daily.max_transactions
-    ) {
-      await triggerCardDeclinedWebhook(
-        cardAuthorizationDeclined,
-        CardAuthorizationDeclineReason.CARD_PRESENT_USE_LIMIT_REACHED_DAILY,
-        person
-      );
-      throw new Error("Daily card_present transaction number limit exceeded");
-    }
+  const { amount: dailyPurchaseLimitAmount, count: dailyPurchaseLimitCount } =
+    getCardLimits(
+      cardData.controls,
+      CardSpendingLimitPeriod.DAILY,
+      DimensionType.PURCHASE
+    );
 
-    if (
-      monthlyLimitAfterAuthorization >
-      cardDetails.cardPresentLimits.monthly.max_amount_cents
-    ) {
-      await triggerCardDeclinedWebhook(
-        cardAuthorizationDeclined,
-        CardAuthorizationDeclineReason.CARD_PRESENT_AMOUNT_LIMIT_REACHED_MONTHLY,
-        person
-      );
-      throw new Error(
-        `Monthly card_present amount limit exceeded (${monthlyLimitAfterAuthorization} > ${cardDetails.cardPresentLimits.monthly.max_amount_cents})`
-      );
-    }
+  const {
+    amount: monthlyPurchaseLimitAmount,
+    count: monthlyPurchaseLimitCount,
+  } = getCardLimits(
+    cardData.controls,
+    CardSpendingLimitPeriod.MONTHLY,
+    DimensionType.PURCHASE
+  );
 
-    if (
-      currentCardUsage.cardPresent.monthly.transactions >
-      cardDetails.cardPresentLimits.monthly.max_transactions
-    ) {
-      await triggerCardDeclinedWebhook(
-        cardAuthorizationDeclined,
-        CardAuthorizationDeclineReason.CARD_PRESENT_USE_LIMIT_REACHED_MONTHLY,
-        person
-      );
-      throw new Error("Monthly card_present transaction number limit exceeded");
-    }
+  if (
+    currentCardUsage.ATM.daily.amount > dailyATMLimitAmount ||
+    currentCardUsage.ATM.daily.count > dailyATMLimitCount ||
+    currentCardUsage.ATM.monthly.amount > monthlyATMLimitAmount ||
+    currentCardUsage.ATM.monthly.count > monthlyATMLimitCount ||
+    currentCardUsage.PURCHASE.daily.amount > dailyPurchaseLimitAmount ||
+    currentCardUsage.PURCHASE.daily.count > dailyPurchaseLimitCount ||
+    currentCardUsage.PURCHASE.monthly.amount > monthlyPurchaseLimitAmount ||
+    currentCardUsage.PURCHASE.monthly.count > monthlyPurchaseLimitCount
+  ) {
+    // triggerCardDeclinedWebhook
```

Comment:
@mbayraktr you will need to trigger new webhook from here with the reason `SPENDING_LIMIT`

Reply:
thank you @lera for the info 🙏